lib/matplotlib/dates.py

```diff
@@ -212,47 +212,31 @@ def _to_ordinalf(dt):
     days, preserving hours, minutes, seconds and microseconds.  Return value
     is a :func:`float`.
     """
-
-    if hasattr(dt, 'tzinfo') and dt.tzinfo is not None:
-        delta = dt.tzinfo.utcoffset(dt)
-        if delta is not None:
-            dt -= delta
+    # Convert to UTC
+    tzi = getattr(dt, 'tzinfo', None)
+    if tzi is not None:
+        dt = dt.astimezone(UTC)
```

Comment:
`date` also doesn't have `tzinfo`, so the `astimezone` line will never fire.

Though, actually, `datetime.datetime.timetz()` returns an object that has `tzinfo` and not `astimezone`. I'm not really sure how to deal with that, though. Are `time` objects an acceptable input here? I think the offset returned will be `None` if you try and get the `utcoffset` of a non-fixed-offset bare `time` object.